### ISSUE TITLE
JabRef 3.0 can't connect to libreoffice

### ISSUE THREAD

**Issue by teertinker** [JabRef/jabref#526]
I use ubuntu 14.04 (64 bit) libreoffice 4.2 and the new version of jabref. 

Neither manually pointing to the correct pathes nor automatic connection does work. Below you can see the output of my error console.
```
INFORMATION: Opening: /home/felix/Dokumente/UniDaten/backup_version_6/backup/publikationen felix.bib
Exception in thread "Spin-0" java.lang.NoClassDefFoundError: spin/off/AWTReflectDispatcherFactory$1
	at spin.off.AWTReflectDispatcherFactory$AWTReflectDispatcher.stop(AWTReflectDispatcherFactory.java:102)
	at spin.off.SpinOffEvaluator$1.run(SpinOffEvaluator.java:110)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: spin.off.AWTReflectDispatcherFactory$1
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 3 more
java.lang.NoClassDefFoundError: com/sun/star/comp/helper/Bootstrap
	at net.sf.jabref.openoffice.OOBibBase.simpleBootstrap(OOBibBase.java:183)
	at net.sf.jabref.openoffice.OOBibBase.<init>(OOBibBase.java:99)
	at net.sf.jabref.openoffice.OpenOfficePanel.run(OpenOfficePanel.java:614)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at spin.Invocation.evaluate(Invocation.java:175)
	at spin.off.SpinOffEvaluator$1.run(SpinOffEvaluator.java:108)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: com.sun.star.comp.helper.Bootstrap
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 10 more
Dez 16, 2015 12:16:25 PM net.sf.jabref.logic.l10n.Localization translate
WARNUNG: Warning: could not get message translation for "Could not connect to running OpenOffice.
Make sure you have installed OpenOffice with Java support.
If connecting manually, please verify program and library paths.

Error message: com/sun/star/comp/helper/Bootstrap" for locale de
```

**Comment by teertinker** [comment]
Update - I upgraded to LibreOffice 5.0 
The problem changes, when doing manual connection. This time jurt.jar is not found. In my installation there is a folder (/usr/lib/libreoffice/program/classes/) where all jar files of libreoffice are stored/linked. Apparently unoil.jar is found, but JabRef looks for jurt.jar in another directory

java.lang.Exception: File not found: /usr/lib/libreoffice/ure-link/share/java/jurt.jar
	at net.sf.jabref.openoffice.OpenOfficePanel.connect(OpenOfficePanel.java:562)
	at net.sf.jabref.openoffice.OpenOfficePanel.access$100(OpenOfficePanel.java:53)
	at net.sf.jabref.openoffice.OpenOfficePanel$3.actionPerformed(OpenOfficePanel.java:217)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
	at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:289)
	at java.awt.Component.processMouseEvent(Component.java:6535)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6300)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4891)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2750)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Dez 16, 2015 1:08:38 PM net.sf.jabref.logic.l10n.Localization translate
WARNUNG: Warning: could not get message translation for "Could not connect to running OpenOffice.
Make sure you have installed OpenOffice with Java support.
If connecting manually, please verify program and library paths.

Error message: File not found: /usr/lib/libreoffice/ure-link/share/java/jurt.jar" for locale de

**Comment by lenhard** [comment]
Thanks for this report. As far as I know, JabRef never featured an integration with Libre Office (so this is not a bug). There is an integration with Open Office (as the error messages say), but there is no guarantee that this works with Libre Office.

Maybe @oscargus can provide some insights?

@JabRef/developers Maybe we should ditch Open Office integration, since Open Office died anyway? I do not know if the current integration can be ported easily to Libre Office

**Comment by oscargus** [comment]
I haven't seen the original issue (as I'm away from a computer), but will
have a look tonight.

It should work with both LibreOffice and OpenOffice and dropping support
for one of them would currently provide a very minor benefit, if any.

The "standard" issue is that JabRef runs in a 32-bit JVM and LO/OO in a
64-bit or vice versa, but no idea if this is the reason here. JabRef has
code to give that hint, but personally I never managed to trigger it
despite trying.

**Comment by teertinker** [comment]
Just two remarks:
- I used to work with LibreOffice and older JabRef versions without any problems (on Linux - automatic connection worked always, on Windows I usually needed to connect manually) - so at least in the past it was not a matter of LO.
- LO and JabRef are on 64 bit

**Comment by oscargus** [comment]
Ah, I see the problem. I changed the default search path for Windows to reflect that the jars were moved, but not on Linux (as I had no idea what had happened there although in hindsight it made sense that they were moved there as well). There's a PR in #530 which should fix it and in 20 minutes or so, there should be a build in http://builds.jabref.org/oolinux/ to try out.

**Comment by teertinker** [comment]
The problem persists in a slightly different way. No path problems are reported but some bootstrap error. See the error message:
```
Exception in thread "Spin-1" java.lang.NoClassDefFoundError: spin/off/AWTReflectDispatcherFactory$1
	at spin.off.AWTReflectDispatcherFactory$AWTReflectDispatcher.stop(AWTReflectDispatcherFactory.java:102)
	at spin.off.SpinOffEvaluator$1.run(SpinOffEvaluator.java:110)
	at java.lang.Thread.run(Thread.java:745)
java.lang.NoClassDefFoundError: com/sun/star/comp/helper/Bootstrap
	at net.sf.jabref.openoffice.OOBibBase.simpleBootstrap(OOBibBase.java:183)
	at net.sf.jabref.openoffice.OOBibBase.<init>(OOBibBase.java:99)
	at net.sf.jabref.openoffice.OpenOfficePanel.run(OpenOfficePanel.java:619)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at spin.Invocation.evaluate(Invocation.java:175)
	at spin.off.SpinOffEvaluator$1.run(SpinOffEvaluator.java:108)
	at java.lang.Thread.run(Thread.java:745)

```

Could there be an issue, that LO is not working properly with JAVA 8 (the only JRE that is currently installed on my system)?

**Comment by oscargus** [comment]
Hmm, no idea really. I only have 8 installed and it works here. This seems to be the same problem as the first time (missing Bootstrap). The Bootstrap method should be in juh.jrt, and I think it would complain earlier if juh.jrt was missing...

**Comment by teertinker** [comment]
There is only a juh.jar in /usr/lib/libreoffice/program/classes...

**Comment by oscargus** [comment]
OK, just to confirm can you run:

`jar tf /usr/lib/libreoffice/program/classes/juh.jar`

**Comment by oscargus** [comment]
Hmm, let me rephrase the previous statement: it doesn't work for me either. Same error. This wasn't the case for 3.0 (which will not work on Linux because of the the incorrect paths). Will need to check what might have changed...

**Comment by oscargus** [comment]
OK! Now it works for me again. Please try the next build (in 20 minutes or so).

**Comment by teertinker** [comment]
Great it works immediately! Thank you so much.
